### ISSUE TITLE
Setup CI/CD pipeline for raising upgrade proposals on SNS.

### ISSUE THREAD

**Issue by ravibazz** [go-bazzinga/hot-or-not-backend-canister#200]
## Description

Develop a Github Pipeline that raises upgrade proposals for the latest change merged into main branch. These are the things that pipeline would perform.

- Get the latest changes merged into main.
- ~Get the version number from the PR label this could be `CI-minor_upddate`, `CI-major_update`, `CI-patch_update`.~
- On tag pushes it generates releases with appropriate tag version else it generates pre-release build with artifacts.
- Once the PR merges based on the label a release should be created with appropriate version along with all the builds of canisters and their respective candid files as artifacts.
- The pipeline should only create builds of necessary canisters according to the changes made on canisters.
- Once the build is complete a proposal should be raised on SNS to make upgrade of canisters using neuron id.

## Tasks

- [ ] build a pipeline that takes PR and generates a release and build artifacts for the release
- [ ] add a mechanism to raise a proposal for the canisters that have changed.
